config/abilities/provider_any.rb

```diff
@@ -31,8 +31,12 @@
 
     if user.has_permission?(:partners)
       can [:show], AccountRelatedEvent do |event|
-        service_id = event.try(:service)&.id || event.try(:service_id)
-        !service_id || user.has_access_to_service?(service_id)
+        services = event.try(:services) || [event.try(:service)].compact
+        service_ids = services.map(&:id) || [event.try(:service_id)].compact
+
+        next true if service_ids.empty? && user.admin?
```

Comment:
if user has not subscribed to any services, then we send notification? Shouldn't we **not** send in fact?

Reply:
We could do it that way if you have a strong opinion and think it's the correct thing to do.

Anyway, old code already notified everyone for account related events that aren't connected to any service:

```ruby
 !service_id || user.has_access_to_service?(service_id)
```

If no service, return `true`; otherwise, return whether the user has access to that service.

In the new code I'm restricting that to only notify admins about events not connected to services.

I think that's fine because an admin could be interested in e.g. new buyers signups even when there is no default service for them. Also, admins and all users can disable any from their notification preferences.

Reply:
But do we want to only notify admins? Correct me if I'm wrong, but I didn't see that in the "complaints list".

Reply:
It's not what the client wants, but we cannot do what the client requests. They want the notifications to be sent to only users with permissions over the affected service, but when an event comes without any attached service, what do we do? previously the notification was being sent to everybody but the client doesn't want that, so the only options are notifying nobody or notifying only admins, which have the option to disable a particular notification if it bothers them.

Reply:
I'm actually unsure about this whole fix, it's not that obvious to me :upside_down_face: 

So, I can see that the permissions for `AccountRelatedEvent` is given only when the user has `:partner` permissions, but for me, if the user can see all Accounts, then why can't it see all notifications about these accounts?

On the other hand, I agree that it might be confusing if a user is notified about someone having subscribed to `Product A`, and they don't even know that this product exists :sweat_smile: 

So, yeah, I guess this is fine.

Reply:
I think that after https://github.com/3scale/porta/pull/4029/commits/afcab73c6b387b8c5eff372caaf6bad4ab2906b8 this line is redundant now, because `user.admin?` implies that `user.has_access_to_all_services?`, so the method will return by then.

Reply:
Right: https://github.com/3scale/porta/pull/4029/commits/4e04b90332c935f1cf8768c0243466d0f7197cc3